analysis/study_definition.py

```diff
@@ -1,13 +1,344 @@
+          "East": 0.1,
+          "London": 0.2,
+          "South East": 0.1,
+          "South West": 0.1,
+        },
+      },
+    },
+  ),
+
+  ## Groups
+  
+  ### Care home - using combination of primis codes and address linkage (based on Schultze report)
+  ### TODO CONFIRM THIS IS CODED CORRECTLY, especially the tpp_care_home_type part ###
+  carehome = patients.satisfying(
+    """
+    carehome_codes
+    OR
+    tpp_care_home_type
+    """,
+
+    return_expectations = {
+      "incidence": 0.05,
+    },
+
+    carehome_codes = patients.with_these_clinical_events(
+      carehome_primis_codes,
+      on_or_before = "index_date",
+      returning = "binary_flag",
+      return_expectations = {"incidence": 0.1}
+    ),
+
+    # COPIED CODE  
+    # tpp_care_home_type=patients.care_home_status_as_of(
+    #  "index_date",
+    #  categorised_as={
+    #    "PC": """
+    #      IsPotentialCareHome
+    #      AND LocationDoesNotRequireNursing='Y'
+    #      AND LocationRequiresNursing='N'
+    #      """,
+    #    "PN": """
+    #      IsPotentialCareHome
+    #      AND LocationDoesNotRequireNursing='N'
+    #      AND LocationRequiresNursing='Y'
+    #      """,
+    #    "PS": "IsPotentialCareHome",
+    #    "U": "DEFAULT",
+    #    },
+    #  return_expectations={
+    #    "rate": "universal",
+    #    "category": {"ratios": {"PC": 0.05, "PN": 0.05, "PS": 0.05, "U": 0.85,},},
+    #    },
+    #),
+
+    ## My attempt to simplify above code to a binary variable
+    tpp_care_home_type=patients.care_home_status_as_of(
+      "index_date",
+      categorised_as={
+        "1": """
+          IsPotentialCareHome
+          """,
+        "0": "DEFAULT",
+        },
+      return_expectations={
+        "rate": "universal",
+        "category": {"ratios": {"1": .15, "0": 0.85,},},
+      },
+    ),
+  ),
+
+  ### Cancer 
+  cancer = patients.with_these_clinical_events(
+    cancer_codes,
+    on_or_before = "index_date",
+    returning = "binary_flag",
+    return_expectations = {"incidence": 0.15}
+  ),
+
+  #####################
+  ## Medication DM&D
+  
+  ## Opioid prescribing
+  ### Any prescribing
+  opioid_any = patients.with_these_medications(
+    opioid_codes,
+    between = ["index_date", "last_day_of_month(index_date)"],
+    returning = "binary_flag",
+    return_expectations = {"incidence": 0.15}
+  ),
+
+  ### High dose prescribing
+  hi_opioid_any = patients.with_these_medications(
+    hi_opioid_codes,
+    between = ["index_date", "last_day_of_month(index_date)"],
+    returning = "binary_flag",
+    return_expectations = {"incidence": 0.05}
+  ),
+
+  ### Any new prescribing (2-year washout)
+  ### TODO: should patients be registered for 2+ years for defining new use?? 
+  opioid_new = patients.satisfying(
+    """
+    opioid_current_date
+    AND 
+    NOT opioid_last_date
+    """, 
+    return_expectations = {
+      "incidence": 0.1,
+    },
+    
+    opioid_current_date = patients.with_these_medications(
```

Comment:
You could combine the creation of opioid_any and opiod_current_date into one chunk of code by adding these two lines to the opiod_any:
include_date_of_match=True,
date_format="YYYY-MM-DD" 
And then a new variable `opiod_any_date` would be created
See documentation [here](https://docs.opensafely.org/study-def-variables/#cohortextractor.patients.registered_as_of)